theano/tensor/signal/downsample.py

```diff
@@ -635,44 +634,70 @@ def c_code(self, node, name, inp, out, sub):
           Py_XDECREF(%(gx)s);
           %(gx)s = (PyArrayObject*) PyArray_ZEROS(4, PyArray_DIMS(%(x)s), x_typenum,0);
         }
```

Comment:
if the output is preallocated region is not initialized to 0., but the code request that.

The old code didn't needed the output to be zeros, as it was putting zeros itself. So it could have allocated not initialized memory. But the new code do not do that, so it should make sure the preallocated output is initialized to 0.